std/file.d

```diff
@@ -5119,3 +5120,87 @@ string tempDir() @trusted
     myFile.write("hello");
     assert(myFile.readText == "hello");
 }
+
+/**
+Returns the available space in a file system.
+
+On `Windows`, this function uses the result of the Windows API function
+$(LINK2 https://msdn.microsoft.com/en-us/library/windows/desktop/aa364937(v=vs.85).aspx, $(D GetDiskFreeSpaceExW)).
+
+On `POSIX` platforms, it uses the $(LINK2 http://man7.org/linux/man-pages/man3/statvfs.3.html,
+$(D statvfs)) function. On `FreeBSD` however, it uses the $(LINK2 https://www.freebsd.org/cgi/man.cgi?query=statfs&sektion=2,
+$(D statfs)) function.
+
+Params:
+    path = a valid path to any file in the file system
+Returns:
+    Available space in bytes
+
+Throws:
+$(LREF FileException) in case of failure
+*/
+ulong getAvailableDiskSpace(string path) @safe
+{
+    import std.exception : enforce;
+    import std.conv : text;
+
+    version (Windows)
+    {
+        import core.sys.windows.winbase : GetDiskFreeSpaceExW, GetLastError;
+        import core.sys.windows.winnt : ULARGE_INTEGER;
+        import std.internal.cstring : tempCStringW;
+
+        ULARGE_INTEGER freeBytesAvailable;
+        auto err = () @trusted {
+            return GetDiskFreeSpaceExW(path.tempCStringW(), &freeBytesAvailable, null, null);
+        } ();
+        enforce(err != 0,
+            new FileException(text("Cannot get available disk space: ", GetLastError())));
```

Comment:
Have a look at the default constructor of FileException:

```
    version(Windows) this(in char[] name,
                          uint errno = .GetLastError(),
                          string file = __FILE__,
                          size_t line = __LINE__) @safe
    {
```

tl;dr: use `cenforce`